### PR TITLE
add more help text for ipfs diag net

### PR DESCRIPTION
**Pull request by whyrusleeping**
Attempts to address issue #666

### CODE REVIEW

File: core/commands/diag.go [hunk 1]
```diff
@@ -56,6 +56,26 @@ var diagNetCmd = &cmds.Command{
 Sends out a message to each node in the network recursively
 requesting a listing of data about them including number of
 connected peers and latencies between them.
+
+The given timeout will be stepped down for each peer the
+request is sent to, to ensure that all peers return in an
+order that will result in the most responses to the initiator.
+The default timeout is 20 seconds.
```
Comment (jbenet) [inline core/commands/diag.go]: how about:
```
The given timeout will be decremented 2s at every network hop, 
ensuring peers try to return their diagnostics before the initiator's 
timeout. If the timeout is too small, some peers may not be reached.
30s and 60s are reasonable timeout values, though network vary.
```

Comment (whyrusleeping) [inline core/commands/diag.go]: So, leave out mention of the default timeout?

File: core/commands/diag.go [hunk 2]
```diff
@@ -56,6 +56,26 @@ var diagNetCmd = &cmds.Command{
...
+
+The 'vis' option may be used to change the output format.
+Currently, two alternate formats are supported, d3, and json.
+The d3 format will output a json object ready to be consumed by
+the chord network viewer, available at the following hash:
```
Comment (jbenet) [inline core/commands/diag.go]:
```
four formats are supported:
 * plain text - easy to read
 * json - the same data as plain text but in json
 * d3 - json ready to be fed into d3view
 * dot - graphviz format (mostly useless)
```

File: core/commands/diag.go [hunk 3]
```diff
@@ -56,6 +56,26 @@ var diagNetCmd = &cmds.Command{
...
+
+    QmbesKpGyQGd5jtJFUGEB1ByPjNFpukhnKZDnkfxUiKn38
```
Comment (jbenet) [inline core/commands/diag.go]:
```
/ipfs/QmbesKpGyQGd5jtJFUGEB1ByPjNFpukhnKZDnkfxUiKn38
```
we should be using paths wherever possible. i can fix `d3view` to do that too

File: core/commands/diag.go (hunk 1 above)
Comment (jbenet) [inline core/commands/diag.go]: oh whoops keep that in

File: core/commands/diag.go (hunk 2 above)
Comment (jbenet) [inline core/commands/diag.go]: (may not need this, was just saying we also have dot :) )

File: core/commands/diag.go (hunk 3 above)
Comment (whyrusleeping) [inline core/commands/diag.go]: alright

Comment (whyrusleeping) [inline core/commands/diag.go]: Should i keep the gateway link in?

File: core/commands/diag.go (hunk 2 above)
Comment (whyrusleeping) [inline core/commands/diag.go]: Oh, did we get rid of the json output?

Comment (jbenet) [inline core/commands/diag.go]: no, it's there

File: core/commands/diag.go (hunk 3 above)
Comment (jbenet) [inline core/commands/diag.go]: yeah!

File: core/commands/diag.go (hunk 2 above)
Comment (whyrusleeping) [inline core/commands/diag.go]: I guess its just not in the `visFmts` array. should it be?

**Comment by jbenet** [comment]
LGTM

**Pull request merged**